src/manifest.c

```diff
@@ -77,323 +77,47 @@ int file_sort_filename_reverse(const void *a, const void *b)
 	return -ret;
 }
 
-static struct manifest *alloc_manifest(int version, char *component)
+static struct manifest *manifest_from_file(int version, char *component, bool header_only, bool is_mix)
 {
-	struct manifest *manifest;
-
-	manifest = calloc(1, sizeof(struct manifest));
-	ON_NULL_ABORT(manifest);
-
-	manifest->version = version;
-	manifest->component = strdup_or_die(component);
-
-	return manifest;
-}
-
-static struct manifest *manifest_from_file(int version, char *component, bool header_only, bool latest, bool is_mix)
-{
-	FILE *infile;
-	char line[MANIFEST_LINE_MAXLEN], *c, *c2;
-	int count = 0;
-	int deleted = 0;
-	int err;
-	struct manifest *manifest;
-	struct list *includes = NULL;
 	char *filename;
 	char *basedir;
-	unsigned long long filecount = 0;
-	unsigned long long contentsize = 0;
-	int manifest_hdr_version;
-	int manifest_enc_version;
+	struct manifest *manifest;
+	struct list *l;
```

Comment:
I would prefer if we use variable names that are more descriptive and longer than one letter, that makes the code more readable.

Reply:
This list is used as an iterator in a for, so I don't think we need a better name, but I could move it to a proper place, just before it's used.